shared/depositutil/deposit.go

```diff
@@ -75,3 +78,43 @@ func WithdrawalCredentialsHash(withdrawalKey bls.SecretKey) []byte {
 	h := hashutil.Hash(withdrawalKey.PublicKey().Marshal())
 	return append([]byte{params.BeaconConfig().BLSWithdrawalPrefixByte}, h[1:]...)[:32]
 }
+
+// VerifyDepositSignature verifies the correctness of Eth1 deposit BLS signature
+func VerifyDepositSignature(dd *ethpb.Deposit_Data) error {
```

Comment:
Is there anywhere else this utility can be used?

Reply:
This method is an exact duplicate of 
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/core/blocks/deposit.go#L222

Reply:
Anyway to refactor that method and only use 1 method throughout the repo.

Reply:
oOoo :( 
which function would you rather keep? the one in blocks/deposit.go or the one I "re-implemented" ?

Reply:
lets keep the one that you have implemented, this is a useful shared method for deposits, so I can see this being used elsewhere. You would just need to have the one in `blocks/deposit.go` use the logic from your method. As long as the logic is only implemented once it should be ok. You can maybe have this chunk in /shared/depositutil
```go
publicKey, err := bls.PublicKeyFromBytes(pub)
	if err != nil {
		return errors.Wrap(err, "could not convert bytes to public key")
	}
	sig, err := bls.SignatureFromBytes(signature)
	if err != nil {
		return errors.Wrap(err, "could not convert bytes to signature")
	}
	root, err := ssz.SigningRoot(obj)
	if err != nil {
		return errors.Wrap(err, "could not get signing root")
	}
	signingData := &pb.SigningData{
		ObjectRoot: root[:],
		Domain:     domain,
	}
	ctrRoot, err := ssz.HashTreeRoot(signingData)
	if err != nil {
		return errors.Wrap(err, "could not get container root")
	}
	if !sig.Verify(publicKey, ctrRoot[:]) {
		return helpers.ErrSigFailedToVerify
	}
	return nil
```
and use that wherever it is needed.

Reply:
Updated @nisdas

Reply:
Updated.